docker/build-image.sh

```diff
@@ -56,42 +58,68 @@ while [[ $# -gt 0 ]]; do
             fi
             shift
             ;;
+        --focal)
+            BUILD_FOCAL=true
+            shift
+            ;;
+        --jammy)
+            BUILD_JAMMY=true
+            shift
+            ;;
     esac
 done
 
 if [[ -z "$COMPONENT_VERSION_STRING" ]]; then
     COMPONENT_VERSION_STRING=$("./get-component-version.sh")
 fi
 
-echo "Building docker image for $IMAGE version: $COMPONENT_VERSION_STRING"
-echo "Final image name: $USERNAME/$IMAGE:$COMPONENT_VERSION_STRING"
+build_image() {
+    local dockerfile_path=$1
+    local tag_suffix=$2
 
-cd ..
-docker build --network=host --no-cache -t $USERNAME/$IMAGE:$COMPONENT_VERSION_STRING \
-    --build-arg VERSION="$COMPONENT_VERSION_STRING" \
-    --build-arg VCS_REF=`git rev-parse --short HEAD` \
-    --build-arg BUILD_DATE=`date -u +”%Y-%m-%dT%H:%M:%SZ”` .
+    echo "Building docker image for $IMAGE version: $COMPONENT_VERSION_STRING using Dockerfile: $dockerfile_path"
+    echo "Final image name: $USERNAME/$IMAGE:$COMPONENT_VERSION_STRING$tag_suffix"
+
+    DOCKER_BUILDKIT=1 docker build --network=host -t $USERNAME/$IMAGE:$COMPONENT_VERSION_STRING$tag_suffix \
+        --build-arg VERSION="$COMPONENT_VERSION_STRING" \
+        --build-arg VCS_REF=`git rev-parse --short HEAD` \
+        --build-arg BUILD_DATE=`date -u +"%Y-%m-%dT%H:%M:%SZ"` \
+        -f $dockerfile_path .
+
+    TAGS+=("$USERNAME/$IMAGE:$COMPONENT_VERSION_STRING$tag_suffix")
+
+    docker tag $USERNAME/$IMAGE:$COMPONENT_VERSION_STRING$tag_suffix $USERNAME/$IMAGE:latest$tag_suffix
+    TAGS+=("$USERNAME/$IMAGE:latest$tag_suffix")
+
+    echo "Tagged $USERNAME/$IMAGE:$COMPONENT_VERSION_STRING$tag_suffix as $USERNAME/$IMAGE:latest$tag_suffix"
+}
 
 TAGS=()
-TAGS+=("$USERNAME/$IMAGE:$COMPONENT_VERSION_STRING")
 
-docker tag $USERNAME/$IMAGE:$COMPONENT_VERSION_STRING $USERNAME/$IMAGE:latest
-TAGS+=("$USERNAME/$IMAGE:latest")
+cd ..
+
+# If neither --focal nor --jammy is specified, build both
```

Comment:
Since we are currently just starting to update the systems to include jammy, it would make more sense to me for this to default to just building focal. Open to discussion on this.

Reply:
That makes sense. I will turn it off by default.